tests/cpp/backends/amdgpu_device_test.cpp

```diff
@@ -0,0 +1,106 @@
+#include "gtest/gtest.h"
+
+#ifdef TI_WITH_AMDGPU
+#include "taichi/ir/ir_builder.h"
+#include "taichi/rhi/amdgpu/amdgpu_driver.h"
+#include "taichi/rhi/amdgpu/amdgpu_context.h"
+#include "taichi/rhi/amdgpu/amdgpu_device.h"
+#include "tests/cpp/program/test_program.h"
+
+namespace taichi {
+namespace lang {
+TEST(AMDGPU, CreateDeviceAndAlloc) {
+  std::unique_ptr<amdgpu::AmdgpuDevice> device =
+      std::make_unique<amdgpu::AmdgpuDevice>();
+  EXPECT_TRUE(device != nullptr);
+  taichi::lang::Device::AllocParams params;
+  params.size = 1048576;
+  params.host_read = false;
+  params.host_write = false;
+  const taichi::lang::DeviceAllocation device_alloc =
+      device->allocate_memory(params);
+
+  // The purpose of the device_alloc_guard is to rule out double free
+  const taichi::lang::DeviceAllocationGuard device_alloc_guard(device_alloc);
+  // Map to CPU, write some values, then check those values
+  void *mapped;
+  EXPECT_EQ(device->map(device_alloc, &mapped), RhiResult::success);
+
+  int *mapped_int = reinterpret_cast<int *>(mapped);
+  for (int i = 0; i < 100; i++) {
+    mapped_int[i] = i;
+  }
+  device->unmap(device_alloc);
+  EXPECT_EQ(device->map(device_alloc, &mapped), RhiResult::success);
+
+  mapped_int = reinterpret_cast<int *>(mapped);
+  for (int i = 0; i < 100; i++) {
+    EXPECT_EQ(mapped_int[i], i);
+  }
+  device->unmap(device_alloc);
+}
+
+TEST(AMDGPU, ImportMemory) {
+  std::unique_ptr<amdgpu::AmdgpuDevice> device =
+      std::make_unique<amdgpu::AmdgpuDevice>();
+  EXPECT_TRUE(device != nullptr);
+
+  int *ptr = nullptr;
+  AMDGPUDriver::get_instance().malloc_managed((void **)&ptr, 400,
+                                              HIP_MEM_ATTACH_GLOBAL);
+  const taichi::lang::DeviceAllocation device_alloc =
+      device->import_memory(ptr, 400);
+
+  for (int i = 0; i < 100; i++) {
+    ptr[i] = i;
+  }
+
+  taichi::lang::Device::AllocParams params;
+  params.size = 400;
+  params.host_read = false;
+  params.host_write = false;
```

Comment:
You'll have to at least set `host_read` to `true` in order to read from the mapped memory. (However, `host_read` isn't implemented nor validated on some backends including CUDA, which is why it still works even if set incorrectly.)